test/blackbox-tests/test-cases/actions/actions-stdxxx-on-success.t

```diff
@@ -0,0 +1,221 @@
+Test for --actions-stdxxx-on-success
+====================================
+
+  $ export BUILD_PATH_PREFIX_MAP="sh=$(which sh):$BUILD_PATH_PREFIX_MAP"
+
+  $ echo '(lang dune 3.0)' > dune-project
+
+  $ cat > dune <<EOF
+  > (rule
+  >  (alias default)
+  >  (action (system "echo 'Hello, world!'")))
+  > 
+  > (rule
+  >  (alias default)
+  >  (action (system "echo 'Something went wrong!' >&2")))
+  > EOF
+
+By default, stdout and stderr are always printed:
+
+  $ dune build
+            sh alias default
+  Hello, world!
+            sh alias default
+  Something went wrong!
+
+swallow tests
+-------------
+
+  $ dune clean
+  $ dune build --actions-stdout-on-success=swallow --actions-stderr-on-success=swallow
+
+must-be-be-empty tests
+----------------------
+
+  $ dune clean
+  $ dune build --actions-stdout-on-success=must-be-empty
+            sh alias default
+  Something went wrong!
+  File "dune", line 1, characters 0-65:
+  1 | (rule
+  2 |  (alias default)
+  3 |  (action (system "echo 'Hello, world!'")))
+            sh alias default (exit 0)
+  (cd _build/default && sh -c 'echo '\''Hello, world!'\''')
+  Hello, world!
+  [1]
+
+  $ dune clean
+  $ dune build --actions-stderr-on-success=must-be-empty
+            sh alias default
+  Hello, world!
+  File "dune", line 5, characters 0-77:
+  5 | (rule
+  6 |  (alias default)
+  7 |  (action (system "echo 'Something went wrong!' >&2")))
+            sh alias default (exit 0)
+  (cd _build/default && sh -c 'echo '\''Something went wrong!'\'' >&2')
+  Something went wrong!
+  [1]
+
+Incremental builds
+------------------
+
+When changing the settings, actions that should be re-executed are
+indeed re-executed:
+
+  $ dune clean
+  $ dune build \
+  >   --actions-stdout-on-success=swallow \
+  >   --actions-stderr-on-success=swallow
+
+  $ dune build
+            sh alias default
+  Hello, world!
+            sh alias default
+  Something went wrong!
```

Comment:
The setting says "swallow" but we print the messages. What went wrong?

Reply:
we don't pass `--actions-stdxxx-on-success=swallow` for this command

Reply:
Here, we are checking that actions are being re-executed if the settings changed.

Reply:
I'm adding a comment about this

Reply:
Oh, sorry, I missed the second `dune build` command. All good then!